### PR TITLE
Use argparse in deepweeds converter to specify paths

### PR DESCRIPTION
**Pull request by jnothman**

### CODE REVIEW

**Comment by jnothman** [comment]
Feel free to do the same for cwfid, @hlydecker

**Comment by hlydecker** [comment]
Good changes; I has been thinking about something like this previously. I've updated the cwfid following your example and it works nicely.

File: conversion_tools/cwfid_to_json/cwfid_to_json.py [hunk 1]
```diff
@@ -20,13 +24,12 @@
...
 # define paths
-# TODO: Adjust this to conform to our chosen file storage structure?
-cwfid_path = pathlib.Path(os.path.realpath('../cwfid_to_json/'))
-
-input_metadata_file = cwfid_path / "labels.csv"
-output_file = cwfid_path / "cwfid_imageinfo.json"
-image_folder = cwfid_path / "cwfid_images"
-dir_name = "cwfid_images"
+ap = argparse.ArgumentParser(description=__doc__)
+ap.add_argument('--labels-dir', default='annotations', type=pathlib.Path)
```
Comment (jnothman) [inline conversion_tools/cwfid_to_json/cwfid_to_json.py:28]: Maybe `--annotations-dir`?

File: conversion_tools/cwfid_to_json/cwfid_to_json.py [hunk 2]
```diff
@@ -20,13 +24,12 @@
...
+ap.add_argument('--image-dir', default='cwfid_images', type=pathlib.Path)
+ap.add_argument('-o', '--out-path', default='cwfid_imageinfo.json', type=pathlib.Path)
+ap.add_argument('--subset-dir', default='.', type=pathlib.Path)
```
Comment (jnothman) [inline conversion_tools/cwfid_to_json/cwfid_to_json.py:31]: I'd probably just reference a single file here `--split-path`

**Review by jnothman — COMMENTED** [review]
Nice. These conversion tools can be specific to the input dataset, so I'd suggest making the CL args more accessible to someone who knows that dataset, rather than universal.

Also, I updated `.github/workflows/validate_json.yml` to make it more explicit for deepweeds conversion.

Otherwise LGTM

**Pull request merged**